Serialization/Editor/ArgsListTests.cs

```diff
@@ -437,6 +438,46 @@ public void ExpectPolymorphicComponentSerializationToBeSupported()
 			Assert.AreEqual(image, resultArgsList[1]);
 		}
 
+		[Test]
+		public void ExpectEnumSerializationToBeSupported()
+		{
+			var argsList = new ArgsList();
+
+			// test that it doesn't throw when specifying the type
+
+			Assert.DoesNotThrow(() => argsList.SetTypes(new List<Type> {typeof(GradientMode)}));
```

Comment:
Minor but would it be worth using your own enum in the test script instead of `GradientMode`?